server.go

```diff
@@ -4100,11 +4206,56 @@ func (s *server) OutboundPeerConnected(connReq *connmgr.ConnReq, conn net.Conn)
 		s.removePeer(connectedPeer)
 		s.ignorePeerTermination[connectedPeer] = struct{}{}
 		s.scheduledPeerConnection[pubStr] = func() {
-			s.peerConnected(conn, connReq, false)
+			s.peerConnected(conn, connReq, false, access)
 		}
 	}
 }
 
+// assignPeerPerms assigns a new peer its permissions.
+func (s *server) assignPeerPerms(remotePub *btcec.PublicKey) (peerAccessStatus,
+	error) {
+
+	// Default is restricted unless the below filters say otherwise.
+	access := peerStatusRestricted
+
+	shouldDisconnect, err := s.authGossiper.ShouldDisconnect(remotePub)
```

Comment:
Add a comment that this queries the banmanager ? Moreover why is the banmanger living in authGossipper and the ProofOfChannel logic in the server ?

Reply:
They should be unified, but I think that should be done in the future and that is why I have structs like `peerScores` as in the future everything can be unified when we have more ban criteria. The gossiper one uses a score and this one doesn't and is very specific to the channel case.